Source/wayland/WaylandServer.m

```diff
@@ -0,0 +1,1482 @@
+	    move = YES;
+	}
+	if (NSEqualPoints(rect.origin, frame.origin) == NO) {
+	    move = YES;
+	}
+
+	wframe = [self _OSFrameToWFrame: rect for: window];
+
+	if (config->pointer.focus &&
+	    config->pointer.focus->window_id == window->window_id) {
+	    config->pointer.y -= (wframe.origin.y - window->pos_y);
+	    config->pointer.x -= (wframe.origin.x - window->pos_x);
+	}
+
+	window->width = wframe.size.width;
+	window->height = wframe.size.height;
+	window->pos_x = wframe.origin.x;
+	window->pos_y = wframe.origin.y;
+
+	xdg_surface_set_window_geometry(window->xdg_surface,
+					window->pos_x,
+					window->pos_y,
+					window->width,
+					window->height);
+/*
+	NSRect flushRect = NSMakeRect(0, 0,
+				      window->width, window->height);
+*/
+	[window->instance flushwindowrect:rect :window->window_id];
+
+	wl_display_dispatch_pending(window->wlconfig->display);
+	wl_display_flush(window->wlconfig->display);
+
+	if (resize == YES) {
+	    NSEvent *ev = [NSEvent otherEventWithType: NSAppKitDefined
+					     location: rect.origin
+					modifierFlags: 0
+					    timestamp: 0
+					 windowNumber: win
+					      context: GSCurrentContext()
+					      subtype: GSAppKitWindowResized
+						data1: rect.size.width
+						data2: rect.size.height];
+	    [(GSWindowWithNumber(window->window_id)) sendEvent: ev];
+	    NSDebugLog(@"placewindow notify resized=%fx%f", rect.size.width, rect.size.height);
+	} else if (move == YES) {
+	    NSEvent *ev = [NSEvent otherEventWithType: NSAppKitDefined
+					     location: NSZeroPoint
+					modifierFlags: 0
+					    timestamp: 0
+					 windowNumber: (int)window->window_id
+					      context: GSCurrentContext()
+					      subtype: GSAppKitWindowMoved
+						data1: rect.origin.x
+						data2: rect.origin.y];
+	    [(GSWindowWithNumber(window->window_id)) sendEvent: ev];
+	    NSDebugLog(@"placewindow notify moved=%fx%f", rect.origin.x, rect.origin.y);
+	}
+
+	NSDebugLog(@"placewindow: newpos=%fx%f", window->pos_x, window->pos_y);
+	NSDebugLog(@"placewindow: newsize=%fx%f", window->width, window->height);
+    }
+}
+
+- (NSRect) windowbounds: (int) win
+{
+    struct window *window = get_window_with_id(wlconfig, win);
+    NSDebugLog(@"windowbounds: win=%d, pos=%dx%d size=%dx%d",
+	       window->window_id, window->pos_x, window->pos_y,
+	       window->width, window->height);
+
+    return NSMakeRect(window->pos_x, window->output->height - window->pos_y,
+		      window->width, window->height);
+}
+
+- (void) setwindowlevel: (int) level : (int) win
+{
+    NSDebugLog(@"setwindowlevel: level=%d win=%d", level, win);
+}
+
+- (int) windowlevel: (int) win
+{
+    NSDebugLog(@"windowlevel: %d", win);
+    return 0;
+}
+
+/** Backends can override this method to return an array of window numbers
+    ordered front to back.  The front most window being the first object
+    in the array.
+    The default implementation returns the visible windows in an
+    unspecified order.
+ */
+- (NSArray *) windowlist
```

Comment:
Why not just use the default implementation?

Reply:
I'll assume original author took `GSDisplayServer` as the base implementation and then implemented individual methods as required. Deleted the useless override.